lib/endpoints/class-wp-rest-terms-controller.php

```diff
@@ -528,6 +489,52 @@ public function delete_item( $request ) {
 	}
 
 	/**
+	 * Prepare a single term for create or update
+	 *
+	 * @param WP_REST_Request $request Request object.
+	 * @return object $prepared_term Term object.
+	 */
+	public function prepare_item_for_database( $request ) {
+		$prepared_term = new stdClass;
+		$prepared_term->args = array();
+
+		if ( isset( $request['name'] ) ) {
+			$prepared_term->name = $request['name'];
+			$prepared_term->args['name'] = $request['name'];
```

Comment:
Er, the `$prepared_term` object should behave similarly to a WP_Term object. Its attributes shouldn't be nested under `$args`.